### PR TITLE
Consistently label App Center icon as “More Apps”

### PR DESCRIPTION
**Pull request by wjt**
https://phabricator.endlessm.com/T28817

### CODE REVIEW

File: js/ui/appDisplay.js [hunk 1]
```diff
@@ -2671,9 +2671,10 @@ class AppCenterIcon extends AppIcon {
         let app = appSys.lookup_app(EOS_APP_CENTER_ID);
 
         super._init(app, viewIconParams, iconParams);
+    }
 
-        this._id = EOS_APP_CENTER_ID;
-        this._name = this.app.get_generic_name();
+    get name() {
+        return _("More Apps");
```
Comment (wjt) [inline js/ui/appDisplay.js]: Doing this this way means that someone (hi) will need to go and copy over all the translations from gnome-software, so I'm prepared to be talked out of it, but I think the decision to show “More Apps” is more a property of our shell than it is of our app center package.

Comment (GeorgesStavracas) [reply js/ui/appDisplay.js]: Alternatively, you can just move the call to `super._init()` to below these lines

**Comment by wjt** [comment]
No, it doesn't work. `AppIcon._init()` is what `super._init()` calls, and it sets `this._name` to the app's non-generic name.

```
diff --git a/js/ui/appDisplay.js b/js/ui/appDisplay.js
index 0a7a5b0a1..b828e7464 100644
--- a/js/ui/appDisplay.js
+++ b/js/ui/appDisplay.js
@@ -1484,6 +1484,7 @@ class ViewIcon extends GObject.Object {
     }
 
     get name() {
+        log(`ViewIcon.get name: this._name = ${this._name}`);
         return this._name;
     }
 
@@ -2190,6 +2191,7 @@ var AppIcon = GObject.registerClass({
         this.app = app;
         this._id = app.get_id();
         this._name = app.get_name();
+        log(`AppIcon._init: this._name = ${this._name}`);
 
         let buttonParams = { button_mask: St.ButtonMask.ONE | St.ButtonMask.TWO };
         iconParams = Params.parse(iconParams, {
@@ -2670,10 +2672,11 @@ class AppCenterIcon extends AppIcon {
         let appSys = Shell.AppSystem.get_default();
         let app = appSys.lookup_app(EOS_APP_CENTER_ID);
 
-        super._init(app, viewIconParams, iconParams);
-
         this._id = EOS_APP_CENTER_ID;
-        this._name = this.app.get_generic_name();
+        this._name = app.get_generic_name();
+        log(`AppCenterIcon._init: this._name = ${this._name}`);
+
+        super._init(app, viewIconParams, iconParams);
     }
 
     _onDragBegin() {
```

Journal output:

```
Nov 29 16:52:56 camille gnome-shell[5395]: AppCenterIcon._init: this._name = More Apps
Nov 29 16:52:56 camille gnome-shell[5395]: AppIcon._init: this._name = App Center
Nov 29 16:52:56 camille gnome-shell[5395]: ViewIcon.get name: this._name = App Center
```

**Comment by GeorgesStavracas** [comment]
Hm, right... I wonder if simply calling `this.icon.label.text = this.name;` would after the `this._name = this.app.get_generic_name();` line would work in that case.

**Comment by wjt** [comment]
It does work. I have opened #605 with that one-line change. Personally I think it is clearer to override the `name` property, but I'm not too familiar with this code.

**Comment by wjt** [comment]
@GeorgesStavracas preferred #605.

**Comment by wjt** [comment]
@GeorgesStavracas preferred #605.

**Pull request closed without merging**